src/main/java/org/aerogear/kryptowire/workflow/KWSubmitStep.java

```diff
@@ -0,0 +1,111 @@
+package org.aerogear.kryptowire.workflow;
+
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.model.TaskListener;
+import jenkins.model.GlobalConfiguration;
+import org.aerogear.kryptowire.GlobalConfigurationImpl;
+import org.aerogear.kryptowire.KryptowireService;
+import org.aerogear.kryptowire.KryptowireServiceImpl;
+import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepDescriptorImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution;
+import org.jenkinsci.plugins.workflow.steps.StepContextParameter;
+import org.json.JSONObject;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+
+public class KWSubmitStep extends AbstractStepImpl {
+
+    private String platform;
+    private String filePath;
+
+    public String getPlatform() {
+        return platform;
+    }
+
+    @DataBoundSetter
+    public void setPlatform(String platform) {
+        this.platform = platform;
+    }
+
+    public String getFilePath() {
+        return filePath;
+    }
+
+    @DataBoundSetter
+    public void setFilePath(String filePath) {
+        this.filePath = filePath;
+    }
+
+    @DataBoundConstructor
+    public KWSubmitStep(@Nonnull String platform, @Nonnull String filePath) {
+        this.platform = platform;
+        this.filePath = filePath;
+    }
+
+    @Extension
+    public static class DescriptorImpl extends AbstractStepDescriptorImpl {
+
+        public DescriptorImpl() {
+            super(KWSubmitExecution.class);
+        }
+
+        @Override
+        public String getFunctionName() {
+            return "kwSubmit";
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "Submit to Kryptowire";
+        }
+
+    }
+
+    public static class KWSubmitExecution extends AbstractSynchronousNonBlockingStepExecution<Void> {
+
+        private static final long serialVersionUID = 1L;
+
+        @Inject
+        transient KWSubmitStep step;
+
+        @StepContextParameter
+        transient TaskListener listener;
+
+        @Override
+        protected Void run() throws Exception {
+
+            GlobalConfigurationImpl pluginConfig = GlobalConfiguration.all().get(GlobalConfigurationImpl.class);
+
+            String kwEndpoint = pluginConfig.getKwEndpoint();
+            String kwApiKey = pluginConfig.getKwApiKey();
+
+            if(StringUtils.isEmpty(kwEndpoint) || StringUtils.isEmpty(kwApiKey)) {
+                throw new RuntimeException("Kryptowire plugin configuration is not set!");
```

Comment:
I think we just need to show a warning message in the log that the submit function will be skipped due to missing plugin config fields (and stop its execution here)

Reply:
I read this to mean for any reason, "build fails if plugin is installed but the submission to Kryptowire fails when the step is invoked from the Jenkins file" https://trello.com/c/aHk4URGz/144-5-as-a-developer-i-want-a-pipeline-step-available-that-allows-me-to-submit-a-binary-to-kryptowire-such-that-i-can-scan-my-binari

tbh, i can't remember what exactly was said about lack of config